app/src/main/java/phone/vishnu/quotes/activity/MainActivity.java

```diff
@@ -714,4 +717,39 @@ private void shareButtonClicked(Context context, int i, Quote q) {
             showShareActionPicker(q);
         }
     }
+
+    private class SwipeListener implements View.OnTouchListener {
+        GestureDetector gestureDetector;
+
+        SwipeListener(View view) {
+            GestureDetector.SimpleOnGestureListener listener =
+                    new GestureDetector.SimpleOnGestureListener() {
+                        @Override
+                        public boolean onDown(MotionEvent e) {
+
+                            homeFAB.startAnimation(
+                                    AnimationUtils.loadAnimation(
+                                            MainActivity.this, R.anim.animate));
+                            if (null == homeFAB.getTag()) {
+                                if (isFABMenuHidden()) {
+                                    openFABMenu();
+                                } else {
+                                    closeFABMenu();
+                                }
+                            } else {
+                                onBackPressed();
+                                resetHomeFAB();
+                            }
```

Comment:
I think triggering the default `onClick()` would be better (with `onClick(homeFAB);`) since this is the exact same code